controllers/routes/routes.js

```diff
@@ -108,21 +108,25 @@ module.exports = function(app) {
     });
   });
 
-  //Update userProfile
+  //Update only modified fields
   app.post('/userProfile/:username/updatedProfile', isLoggedIn, function (req, res) {
-    User.findOne({ username: req.user.username }, function (err, currentUser) {
-      if (err) {
-        res.redirect('/userProfile/:username');
-      } else {
-        currentUser.profileData.firstLastName = req.body.firstLastName;
-        currentUser.profileData.bio = req.body.bio;
-        currentUser.profileData.location = req.body.location;
-        currentUser.profileData.website = req.body.website;
-        currentUser.profileData.birthdate = req.body.birthdate;
-        currentUser.save()
-        res.redirect('/timeline/' + currentUser.username);
-      }
-    });
+    User.findOneAndUpdate({ username: req.user.username },
+      {
+        $set: {
+          "profileData.firstLastName": req.body.firstLastName,
+          "profileData.bio": req.body.bio,
+          "profileData.location": req.body.location,
+          "profileData.website": req.body.website,
+          "profileData.birthdate": req.body.birthdate,
+        }
+      },
+      function (err, currentUser) {
+        if (err) {
+          res.redirect('/userProfile/:username');
```

Comment:
Instead of `:username`, it'll probably have to `currentUser.username` like the line below